numpy/core/tests/test_numeric.py

```diff
@@ -2848,6 +2864,13 @@ def test_ip_all_isclose(self):
         for (x, y) in self.all_close_tests:
             self.tst_all_isclose(x, y)
 
+        x = np.array([2.3, 3.6, 4.4, np.nan])
+        y = np.array([2, 3, 4, np.nan])
+        atol = [0.31, 0, 0, 1]
+        rtol = [0, 0.21, 0.11, 1]
+        assert np.allclose(x, y, atol=atol, rtol=rtol, equal_nan=True)
+        assert not np.allclose(x, y, atol=0.1, rtol=0.1, equal_nan=True)
```

Comment:
I feel like there should be a verbatim test of the original issue. Should it be here or in a new test, e.g.
`test_allclose_gh14320`?